xtask/src/cross.rs

```diff
@@ -0,0 +1,59 @@
+use anyhow::Result;
+use clap::Parser;
+use xshell::{cmd, Shell};
+
+#[derive(Debug, Parser)]
+pub(crate) struct Options {
+    /// Target architecture
+    #[clap(long, default_value = "x86_64-unknown-linux-musl")]
+    pub(crate) target: String,
+
+    /// Binary to compile
+    #[clap(long, default_value = "test-suite")]
+    pub(crate) binary: String,
+
+    #[clap(subcommand)]
+    command: Command,
+}
+
+#[derive(Debug, Parser)]
+enum Command {
+    /// Build binary and copy it to destination folder
+    Build {
+        /// Where to copy the file
+        #[clap(long, default_value = "/tmp/")]
+        destination: String,
+    },
+    /// Run over the given SSH connection.
+    Run {
+        /// Target ssh
+        #[clap(long, default_value = "qemu")]
+        ssh_target: String,
+        /// Arguments passed on process execution
+        args: Vec<String>,
+    },
+}
+
+pub(crate) fn run(options: Options) -> Result<()> {
+    let sh = Shell::new()?;
+    let Options {
+        target,
+        binary,
+        command,
+    } = options;
+    cmd!(
+        sh,
+        "cross build --target {target} --target-dir target/cross --workspace --bin {binary}"
```

Comment:
I think it's good enough since the target user is a developer. If we add the check we could explain what each command is used for. What do you think?
```
matteo@gastly:~/projects/pulsar$ cargo xtask cross run
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/xtask cross run`
$ cross build --target x86_64-unknown-linux-musl --target-dir target/cross --workspace --bin test-suite
command not found: `cross`
```

Reply:
it's fine